lib/pbench/server/api/resources/compare_dataset.py

```diff
@@ -0,0 +1,145 @@
+from http import HTTPStatus
+from urllib.request import Request
+
+from flask import current_app, jsonify
+from flask.wrappers import Response
+from pquisby.lib.post_processing import BenchmarkName, InputType, QuisbyProcessing
+
+from pbench.server import OperationCode, PbenchServerConfig
+from pbench.server.api.resources import (
+    APIAbort,
+    ApiAuthorizationType,
+    ApiBase,
+    ApiContext,
+    APIInternalError,
+    ApiMethod,
+    ApiParams,
+    ApiSchema,
+    convert_dataset,
+    Parameter,
+    ParamType,
+    Schema,
+    UnauthorizedAccess,
+)
+import pbench.server.auth.auth as Auth
+from pbench.server.cache_manager import (
+    CacheManager,
+    TarballNotFound,
+    TarballUnpackError,
+)
+from pbench.server.database import Dataset
+from pbench.server.database.models.users import User
+
+
+class CompareDataset(ApiBase):
+    """
+    This class implements the Server API used to retrieve comparison data for visualization.
+    """
+
+    def __init__(self, config: PbenchServerConfig):
+        super().__init__(
+            config,
+            ApiSchema(
+                ApiMethod.GET,
+                OperationCode.READ,
+                query_schema=Schema(
+                    Parameter(
+                        "datasets", ParamType.LIST, string_list=",", required=True
+                    ),
+                ),
+                authorization=ApiAuthorizationType.NONE,
+            ),
+        )
+
+    def _get(
+        self, params: ApiParams, request: Request, context: ApiContext
+    ) -> Response:
+        """
+        This function is using Quisby to compare results into a form that supports visualization
+
+        Args:
+            params: includes the uri parameters, which provide the list of dataset.
+            request: Original incoming Request object
+            context: API context dictionary
+
+        Raises:
+            UnauthorizedAccess : The user isn't authorized for the requestedaccess.
+            APIAbort, reporting "NOT_FOUND" and "INTERNAL_SERVER_ERROR"
+            APIInternalError, reporting the failure message
+
+        GET /api/v1/compare?datasets=d1,d2,d3
+        """
+
+        datasets = params.query.get("datasets")
+        authorized_user: User = Auth.token_auth.current_user()
+
+        if not authorized_user:
+            raise APIAbort(
+                HTTPStatus.UNAUTHORIZED,
+                "User provided access_token is invalid or expired",
+            )
+
+        ds_list = [convert_dataset(ds, None) for ds in datasets]
+        cache_m = CacheManager(self.config, current_app.logger)
+
+        benchmark_check = []
```

Comment:
`benchmark_check = set()` will create a set in the first place. You don't need to accumulate the entire list first and then convert! (Use `benchmark_check.add()` instead of `benchmark_check.append()`.)

Also, at the end, you could consider adding the set (or `",".join(set)`) to the `APIAbort` message giving the list of benchmark types they specified...